src/panels/config/automation/add-automation-element/ha-automation-add-from-target.ts

```diff
@@ -0,0 +1,1609 @@
+import "@home-assistant/webawesome/dist/components/tree-item/tree-item";
+import "@home-assistant/webawesome/dist/components/tree/tree";
+import type { WaSelectionChangeEvent } from "@home-assistant/webawesome/dist/events/selection-change";
+import { consume } from "@lit/context";
+import { mdiTextureBox } from "@mdi/js";
+import type { HassEntity } from "home-assistant-js-websocket";
+import {
+  css,
+  html,
+  LitElement,
+  nothing,
+  type PropertyValues,
+  type TemplateResult,
+} from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { ifDefined } from "lit/directives/if-defined";
+import memoizeOne from "memoize-one";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { computeAreaName } from "../../../../common/entity/compute_area_name";
+import { computeDeviceName } from "../../../../common/entity/compute_device_name";
+import { computeEntityNameList } from "../../../../common/entity/compute_entity_name_display";
+import { stringCompare } from "../../../../common/string/compare";
+import "../../../../components/entity/state-badge";
+import "../../../../components/ha-floor-icon";
+import "../../../../components/ha-icon";
+import "../../../../components/ha-icon-next";
+import "../../../../components/ha-md-list";
+import "../../../../components/ha-md-list-item";
+import "../../../../components/ha-section-title";
+import "../../../../components/ha-svg-icon";
+import {
+  getAreasNestedInFloors,
+  type AreaFloorValue,
+  type FloorComboBoxItem,
+  type FloorNestedComboBoxItem,
+  type UnassignedAreasFloorComboBoxItem,
+} from "../../../../data/area_floor";
+import {
+  getAreaDeviceLookup,
+  getAreaEntityLookup,
+} from "../../../../data/area_registry";
+import {
+  getConfigEntries,
+  type ConfigEntry,
+} from "../../../../data/config_entries";
+import {
+  areasContext,
+  devicesContext,
+  entitiesContext,
+  floorsContext,
+  labelsContext,
+  localizeContext,
+  statesContext,
+} from "../../../../data/context";
+import { getDeviceEntityLookup } from "../../../../data/device_registry";
+import {
+  domainToName,
+  type DomainManifestLookup,
+} from "../../../../data/integration";
+import {
+  getLabels,
+  type LabelRegistryEntry,
+} from "../../../../data/label_registry";
+import {
+  TARGET_SEPARATOR,
+  type SingleHassServiceTarget,
+} from "../../../../data/target";
+import type { HomeAssistant } from "../../../../types";
+import { brandsUrl } from "../../../../util/brands-url";
+
+interface Level1Entries {
```

Comment:
Maybe we can give these clearer names? As `Level1` is just for floor children right?

Reply:
No, because Devices, services, entities and helpers are on level1 too.